app/javascript/controllers/form_date_controller.js

```diff
@@ -1,7 +1,7 @@
 import { Controller } from '@hotwired/stimulus'
 
 export default class extends Controller {
-  static targets = ['year', 'month', 'day']
+  static targets = ['year', 'month', 'day', 'approximate']
```

Comment:
Adding support for approximate. Having a day and approximate are mutually exclusive.